### ISSUE TITLE
[1.1.9] Disabling TMC2130 Sensorless Homing for Z Axis

### ISSUE THREAD

**Issue by mbuc** [MarlinFirmware/Marlin#11535]
### Description

Documentation and comments do not provide adequate instructions for disabling sensorless homing on a per-axis basis.

### Workaround

If, like me, you are using a traditional homing switch to home your Z axis to Z-min you may come across the following build errors from sanity check:

```
SENSORLESS_HOMING requires Z_MIN_ENDSTOP_INVERTING and ENDSTOPPULLUP_ZMIN when homing to Z_MIN.
```

But I don't want to use sensorless homing on Z, and I'm not using a probe so it's not disabled automatically.

My workaround is to comment out #define Z_HOMING_SENSITIVITY:

```cpp
//#define Z_HOMING_SENSITIVITY 8
```

Is there something I missed?

**Comment by teemuatlut** [comment]
> Is there something I missed?

Nope, you got it right. I think I will be disabling `Z_HOMING_SENSITIVITY` by default anyway.

**Comment by jneilliii** [comment]
Is this true also for 2.0.0?  I just installed 2.0.0 bugfix and have FIX_MOUNTED_PROBE enabled, SENSORLESS_HOMING enabled and commented out the Z_HOMING_SENSITIVITY and when I try to home Z my nozzle rams into the bed,  M119 reports the Z does get triggered and Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN is enabled.  I'm using a [mini-ir sensor](https://miscsolutions.wordpress.com/mini-height-sensor-board/) and my previous 1.1.9 build was working fine.

Here are my configs if it helps. 

[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/2300202/Marlin.zip)

**Comment by thinkyhead** [comment]
You may want to triple-check with `M119` that your probe is triggering when it should. When doing probe, Marlin moves down until it sees that the probe is triggered. If Marlin doesn't stop, this indicates that it doesn't see that the probe is triggered.

**Comment by jneilliii** [comment]
I will reflash 2.0.0 and triple check. I know I double checked that it was showing triggered with M119. The good news is that 2.0.0 fixed my full graphics controller button not working in 1.1.9.

**Comment by jneilliii** [comment]
BTW, there were some extra tweaks I had to do for the Einsy Rambo board to work with the full graphics controller. I added the same if mb(einsy_board) conditions being applied for minirambo.

**Comment by thinkyhead** [comment]
> BTW, there were some extra tweaks I had to do for the Einsy Rambo board to work with the full graphics controller. I added the same if mb(einsy_board) conditions being applied for minirambo.

Good work! Looking forward to your patch submission.

**Comment by jneilliii** [comment]
Ok, so I don't know what happened.  To verify I recompiled and flashed to my Einsy board again with the previous configuration.  I reset to hardcoded default settings and then run G28, all is good and M119 reports a triggered event on Z with my mini-ir sensor when the led on it lights up.  I start a G29 P1 for UBL and the nozzle moves to the first probe point and then goes up and down a bunch of times reporting `Error:Probing failed` at every down position.  This goes for what I assume to be some timeout/attempt value somewhere. If I enter a G28 after that the nozzle will ram into the bed when homing Z position. It also is not reporting triggered for Z when I manually move the nozzle down until the mini-ir led indicates a triggered event. I don't recall having the Probing failed issue before when I initially tested as I used my OctoPrint-BedLevelVisualizer plugin to look at the mesh, so maybe something related to storing settings as mentioned in #11043 is the root cause?

**Comment by jneilliii** [comment]
I just confirmed with a fresh pull of bugfix 2.0.x and compile/flash. Running G28 after M502 works as expected and homes and triggers.  If I run a G29 P1 I get the probing failed errors and the end stop is not triggering at any position. Once the timeout occurs Z remains in an open position even though the led on the sensor indicates a trigger until a power cycle occurs and then Z triggers again until you run a UBL command.

**Comment by jneilliii** [comment]
I think I found my issue. My version of 1.1.9 was an older pre-release bugfix version before the trinamic changes in configuration_adv.h (MINIMUM_STEPPER_DIR_DELAY, MINIMUM_STEPPER_PULSE and MAXIMUM_STEPPER_RATE) which I missed.  After going through those sections I think I might be working as expected now, at least in the official release of 1.1.9.

**Comment by jneilliii** [comment]
So it seems no love for me with 2.0.x with the same settings that seem to work fine in official release 1.1.9 and my mini-ir probe.  It seems to continually have probe failed errors.  With the official release 1.1.9 that happens occasionally on probing where the probe seems to get some sort of power feedback when making certain kinds of moves around the bed and the sensor ends up resetting itself and the next probe attempt would succeed.  This was never an issue with my previous bugfix.1.1.x version (info below).

```
Revision: c76a466ba96f325d4d5050e44743e3d87d82f663
Author: Scott Lahteine <thinkyhead@users.noreply.github.com>
Date: 5/22/2018 3:52:24 PM
Message:
Followup to endstops update
```

**Comment by thinkyhead** [comment]
Try this:

- Download Marlin `bugfix-2.0.x` to test with the latest code.
- Enable `DEBUG_LEVELING_FEATURE` and `M114_DETAIL` and re-flash.
- Connect to your printer from host software such as Cura, Printrun or Repetier Host.
- Issue the command `M111 S247` to enable maximum logging.
- Perform a `G28` to do your standard homing procedure.
- Do a `G29` to probe the bed.
- Examine the log output to see if it reveals anything more about the failure.

**Comment by jneilliii** [comment]
Must have been a bad flash/fluke Scott.  Thanks for the help, bugfix-2.0.x appears to be working fine after re-compiling and flashing.  The one thing I did do that I didn't before was to manually clear out Arduino's output directory.  I did the workaround to point the output directory to c:\arduino to get around the length compile error and maybe there were some remnants left over from the previous compile that was being used maybe?  Don't know, but the bugfix-2.0.x downloaded today seems to be working as expected.
